File: LEETCODE/LEETCODE-PYTHON/342.py
# 342. Power of Four
# https://leetcode.com/problems/power-of-four/

#Method-1 Brute Force
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n<=0:
            return False

        while n%4==0:
            n//=4

        return n==1
        
#Method-2 using log
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0:
            return False
    
        x = math.log(n, 4) 
        return math.ceil(x) == math.floor(x)
        

File: LEETCODE/LEETCODE-JAVA/342.java
// 342. Power of Four
// https://leetcode.com/problems/power-of-four/

//Method -1 Brute force
class Solution {
    public boolean isPowerOfFour(int n) {
        if(n<=0){
            return false;
        }

        while(n%4==0){
            n/=4;
        }

        return n==1;
    }
}

//Method-2 using log
class Solution {
    public boolean isPowerOfFour(int n) {
        if (n <= 0) {
            return false;
        }

        double x = Math.log(n) / Math.log(4);

        return Math.ceil(x) == Math.floor(x);
    }
}
